aws/data_source_aws_api_gateway_resource_test.go

```diff
@@ -0,0 +1,100 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestDataSourceAwsApiGatewayResource(t *testing.T) {
+	rName := acctest.RandString(8)
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsApiGatewayResourceConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceAwsApiGatewayResourceCheck("aws_api_gateway_resource.example_v1"),
+					testAccDataSourceAwsApiGatewayResourceCheck("aws_api_gateway_resource.example_v1_endpoint"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAwsApiGatewayResourceCheck(name string) resource.TestCheckFunc {
```

Comment:
We can skip creating this additional function by using the [`resource.TestCheckResourceAttrPair()`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckResourceAttrPair) helper function, e.g.

```go
resourceName := "aws_api_gateway_resource.example_v1"
dataSourceName := "data.aws_api_gateway_resource.example_v1"
// ...
resource.TestCheckResourceAttrPair(resourceName, "id", dataSourceName, "id"),
resource.TestCheckResourceAttrPair(resourceName, "parent_id", dataSourceName, "parent_id"),
resource.TestCheckResourceAttrPair(resourceName, "path_part", dataSourceName, "path_part"),
```